addons/website/models/website.py

```diff
@@ -1460,6 +1460,8 @@ def _is_canonical_url(self):
     @tools.ormcache('self.id')
     def _get_cached_values(self):
         self.ensure_one()
+        # Manual prefetch to avoid loading any translatable field, as the context lang might be invalid
+        self.fetch(['user_id', 'company_id', 'default_lang_id', 'homepage_url'])
```

Comment:
- Sorry to be annoying but we try to enforce 80 char per line for comments in the website app :pray: 
- I am really not an ORM expert: not sure to have fully understood why this is particularly needed here and not everywhere else but I don't really need to. However, maybe add a bit more in the comment here if possible? Maybe your whole paragraph:
```
However, ir.http:_match is called at a time when the environment hasn't
been completely initialized (it will be properly initialized when
ir.http:_authenticate is called), and its context language hasn't been
checked against activated languages yet.
```
Your commit message seems good, the code would benefit having the relevant part of it as a comment too IMO :)